subsys/net/ip/net_stats.h

```diff
@@ -340,6 +340,21 @@ static inline void net_stats_update_tx_time(struct net_if *iface,
 }
 #endif /* CONFIG_NET_CONTEXT_TIMESTAMP && STATISTICS */
 
+#if defined(CONFIG_NET_PKT_RXTIME_STATS) && defined(CONFIG_NET_STATISTICS)
+static inline void net_stats_update_rx_time(struct net_if *iface,
+					    u32_t start_time,
+					    u32_t end_time)
+{
+	u32_t diff = abs(end_time - start_time);
```

Comment:
This is incorrect way to calculate time difference. (Standard comment to your code like that, eh?)

Reply:
> (Standard comment to your code like that, eh?)

That was also a standard comment from you, eh?

Reply:
No, I multiple times suggested, that when dealing with times represented as unsigned integers, one of them known to be physically after another, then `after - before` is the correct way to calculate the duration, including supplying a Linux test sample which shows correctness for boundary cases, etc. So, it's me who got used to your ignoring the review comments and shoving problematic code into the tree, due to which attitude the network stack stays broken for years.

Reply:
Instead of whining about this, just propose a one liner that does what you expect.
The code currently in the PR works just fine, just tested it in Linux.

Reply:
Please be ready to accept criticism in you PRs without calling it whining. (And yeah, my point is that this issue was raised enough times to be remembered, but maybe it's just me.)

The correct code is `u32_t diff = end_time - start_time;`.

> The code currently in the PR works just fine, just tested it in Linux.

Well, so what your Linux calculates for the case of start_time = 2, end_time = 1 ?

Reply:
Yep, good point. You could have proposed a proper fix in your first comment so we could avoid this ping pong.

Reply:
> You could have proposed a proper fix in your first comment so we could avoid this ping pong.

Noted, thanks. Will try to look at this stuff as if it is new and doesn't have (up to) 3 year-history behind it.